pyproject.toml

```diff
@@ -19,6 +19,8 @@ exclude = '''
 
 [tool.ruff]
 target-version = "py311"
```

Comment:
@ogajduse just a side comment not regarding these changes, shouldn't we change the ruff target version to `py312` when we already use 3.12 in our frameworks, same on line 8 in tool.black target-version

Reply:
We should. That should be done, however, every time we bump the Python version. AFAIK there are no guidelines or documentation, making it easy to forget.

Reply:
Looking at https://docs.astral.sh/ruff/configuration, `target-version` is `The minimum Python version that should be supported`. Therefore it should be equal to the minimal version that we support. We still support py310 - defined by the GH Actions matrix.